pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1857,7 +1857,7 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(ctx context.Context, clusterName
 	}
 
 	// delete the loadbalancer and all its sub-resources.
-	if lbaas.opts.UseOctavia {
+	if lbaas.opts.UseOctavia && lbaas.opts.CascadeDelete {
```

Comment:
@hamzazafar @lingxiankong  one question here, IIUC if cascadeDelete is false , dont we suppose to proceed with this operation with deleteOpts := loadbalancers.DeleteOpts{Cascade: false} ?
something like this instead
`if lbaas.opts.UseOctavia`
` deleteOpts := loadbalancers.DeleteOpts{Cascade: lbaas.opts.CascadeDelete} `

Reply:
@ramineni if we set `Cascade: false` then only the load balancer resource will get deleted, listeners, health-monitors, pools and members will be left behind. 

In current implementation, if cascade delete is false, then load balancer and related resources are deleted one by one [ref](https://github.com/kubernetes/cloud-provider-openstack/blob/3c1dcc8b7083e0d6d0749cb003cff3df7a468fad/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go#L1865).

Reply:
ok. Got it